lib/helper.ts

```diff
@@ -162,35 +166,47 @@ export function getFormat(
 export function buildRequestFileObject(
   fileParam: FileWithMetadata
 ): FileObject {
+  let fileObj: FileObject;
+  if (isFileObject(fileParam.data)) {
+    // For backward compatibility, we allow the data to be a FileObject.
+    fileObj = { value: fileParam.data.value, options: {} };
+    if (fileParam.data.options) {
+      fileObj.options = {
+        filename: fileParam.filename || fileParam.data.options.filename,
+        contentType: fileParam.contentType || fileParam.data.options.contentType,
+      }
+    }
+  } else {
+    fileObj = {
+      value: fileParam.data,
+      options: {
+        filename: fileParam.filename,
+        contentType: fileParam.contentType,
+      }
+     };
+  }
+
+  // Also for backward compatibility, we allow data to be a string
+  if (typeof fileObj.value === 'string') {
+    fileObj.value = Buffer.from(fileObj.value);
+  }
+
   // build filename
-  let filename: string | Buffer;
-  if (fileParam.filename) {
-    // prioritize user-given filenames
-    filename = fileParam.filename;
-  } else if (isFileStream(fileParam.data)) {
+  let filename: string | Buffer = fileObj.options.filename;
+  if (!filename && isFileStream(fileObj.value)) {
     // if readable stream with path property
-    filename = fileParam.data.path;
+    filename = fileObj.value.path;
   }
   // toString handles the case when path is a buffer
-  filename = filename ? basename(filename.toString()) : '_';
+  fileObj.options.filename = filename ? basename(filename.toString()) : '_';
 
   // build contentType
-  let contentType: string = 'application/octet-stream';
-  if (fileParam.contentType) {
-    // prioritize user-given content_type
-    contentType = fileParam.contentType;
-  } else {
-    // else utilize file-type package
-    contentType = getContentType(fileParam.data) || contentType;
+  if (!fileObj.options.contentType && isFileData(fileObj.value)) {
+
```

Comment:
This is picky I know, but can we remove this newline? The TS linter won't complain about but technically it violates the style guide